src/hdmf/utils.py

```diff
@@ -121,6 +125,51 @@ def __format_type(argtype):
         raise ValueError("argtype must be a type, str, list, or tuple")
 
 
+def __check_enum_argtype(argtype):
+    """Return True/False whether the given argtype is a supported docval enum type (bool, ints, floats, str)"""
+    if isinstance(argtype, (list, tuple)):
+        for i in argtype:
+            if not __check_enum_argtype(i):
+                return False
+        return True
+    return argtype in __supported_bool_types + __supported_int_types + __supported_float_types + (str, )
```

Comment:
I'm a bit confused by this function. Why is it recursive? We already have the line 148 that makes sure we have a list or a tuple. I also think we should perform a check that the enum values match the indicated `dtype` arg.

Reply:
This checks whether the types listed under the `'type': (bool, int)` or `'type': str` part of the docval argument are allowed for enum validation. If the argument allows a `TimeSeries` to be passed in, for example, this check would fail. I agree - this does not need to be recursive. I'll simplify it. I just wanted to handle both the case where a type is passed in and a list of types is passed in.

Line 148 checks whether the values under the `'enum': (True, 1)` is a list/tuple (it would not make sense to have only one allowed value).